### PR TITLE
Adxl313 work

### PR DESCRIPTION
**Pull request by danmois**
# Description

Addition of support for ADXL312, ADXL313, and ADXL314. 

The creation of the class corresponding to the ADXL313 driver and the example script corresponding to the same driver.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] examples/adxl313_example.py

**Test Configuration**:
* Hardware: Nucleo-F401RE + EVAL-ADXL312Z, EVAL-ADXL313Z, and EVAL-ADXL314Z development boards
* OS: no-OS running an iio server exposing a basic ADXL312, ADXL313, and ADXL314 device iio device

# Documentation

The documentation has been updated (doc/source/devices/adi.adxl313.rst created, and updates of doc/source/devices/index.rst and supported_parts.md).

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by danmois** [comment]
V2

- Codespell recommended corrections

File: adi/adxl313.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        ]
+
+        self._ctrl = None
+
+        if device_name not in compatible_parts:
+            raise Exception("Not a compatible device: " + device_name)
+
+        # Select the device matching device_name as working device
+        for device in self._ctx.devices:
+            print("Found device {}".format(device.name))
```
Comment (adrimbarean) [inline adi/adxl313.py:66]: nitpick: Probably not gating, but if this lists all devices present in a context I think it's too verbose for a part class. We can just say when we found the device and maybe if we didn't.

File: adi/adxl313.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            return self._get_iio_attr(self.name, "sampling_frequency_available", False)
+
+        @property
+        def range(self):
+            """ADXL313 per-channel range."""
+            return self._get_iio_attr_str(self.name, "range", False)
+
+        @range.setter
+        def range(self, value):
+            print(value)
```
Comment (adrimbarean) [inline adi/adxl313.py:121]: nitpick: if we didn't use this for the rest of the properties, we should remain consistent.

File: doc/source/devices/index.rst [hunk 3]
```diff
@@ -44,15 +44,16 @@ Supported Devices
    adi.adis16507
    adi.adl5960
    adi.admv8818
-   adi.adpd188
    adi.adpd1080
+   adi.adpd188
```
Comment (adrimbarean) [inline doc/source/devices/index.rst:48]: Is this intended?

File: examples/adxl313_example.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+# myacc = adi.adxl313(uri="ip:192.168.1.10", device_name="ADXL313")
+myacc.rx_buffer_size = 32
+myacc.rx_enabled_channels = [0, 1, 2]
+
+sfa = myacc.accel_x.sampling_frequency_available
+print("\nSample frequencies available:")
+print(sfa)
+
+print("\nInitial sample frequency:")
+print("Sample frequency: " + str(myacc.accel_x.sampling_frequency))
```
Comment (adrimbarean) [inline examples/adxl313_example.py:56]: Is the sampling frequency a per device or per channel attribute? If it's a per device attribute maybe it's worth treating it as such, but that may entail a change of the driver, so this is just something to think about.

**Review by adrimbarean — COMMENTED** [review]
Solid work and a good addition to our roster. A few questions and comments from me.

File: doc/source/devices/index.rst (hunk 3 above)
Comment (danmois) [reply doc/source/devices/index.rst:48]: This was not intended, but the reordering was performed by the automatic tools.

File: examples/adxl313_example.py (hunk 4 above)
Comment (danmois) [reply examples/adxl313_example.py:56]: It is an channel attribute, that is shared between all channels. Once changed for one channel, it is valid for all the other ones. I used the same pattern as in other drivers and classes.

File: adi/adxl313.py (hunk 2 above)
Comment (danmois) [reply adi/adxl313.py:121]: Thank you for the observation. This line will be removed, since it remained from the development stage.

**Review by tfcollins — CHANGES_REQUESTED** [review]
Please add some basic tests. A good starting point: https://github.com/analogdevicesinc/pyadi-iio/blob/master/test/test_adxl355.py

-Travis

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Pull request merged**